### ISSUE TITLE
ci: Suppress API breaking changes warning

### ISSUE THREAD

**Issue by storopoli** [rust-bitcoin/rust-bitcoin#3082]
I'm kinda out of the loop on this but I might read it more later.

> API BREAKING CHANGE DETECTED

I much prefer the alerts vs the CI breakage and the diff being stuffed into git.  Although I wish there was a way to silence it after the first time or something because it gets pretty noisy after every commit.

_Originally posted by @yancyribbens in https://github.com/rust-bitcoin/rust-bitcoin/issues/3003#issuecomment-2235221506_

We could further introspect the PR with `actions/github-script` to suppress the warning if we already commented on the PR.

**Comment by storopoli** [comment]
Do we really need this? @coveralls already spams PRs on every new (force)push as well?

**Comment by apoelstra** [comment]
I don't really mind it. Though it is pretty noisy.

I would prioritize trying to get it to post more useful information before I prioritized suppressing it.

**Comment by Kixunil** [comment]
We don't need twice as much span though.

**Comment by tcharding** [comment]
OT, I have yet to read the output of coveralls more than a couple times and find it annoyingly spammy. On topic, I would prefer at least if the API break message was not in all caps. Unless there is some reason for the message I rekon the label is actually enough.

**Comment by Kixunil** [comment]
Same about coverall, I ignore it almost every time.  Label might be enough, though I'd still enjoy seeing the message once since "bot added a label" isn't that visible.

**Comment by storopoli** [comment]
About coveralls, in their [README](https://github.com/coverallsapp/github-action?tab=readme-ov-file#coveralls-comments-arent-added-to-my-pull-request):

> You have enabled the "LEAVE COMMENTS?" setting in the "Pull Request Alerts" area in your Repo Settings page inside the Coveralls app.

Hence, we can turn the `LEAVE COMMENTS` off.

**Comment by apoelstra** [comment]
With coveralls I'd kinda like if it edited the main PR description so that the text would wind up in the git commit log. Though I imagine this would annoy people and also have weird glitches and race conditions with manual edits.

I wouldn't mind turning both bots off -- if we turned "leave comments" off on coveralls would it still be possible to see coverage changes, if we looked for them?

**Comment by storopoli** [comment]
> if we turned "leave comments" off on coveralls would it still be possible to see coverage changes, if we looked for them?

I think so. It might be worth a try.
I will try to hack the API break comments to only comment once.

**Comment by Kixunil** [comment]
@apoelstra I believe they did it recently but I didn't pay attention enough to be sure. But the thing is I tend to ignore even the first one.

**Comment by tcharding** [comment]
> But the thing is I tend to ignore even the first one.

Is it adding any value then? I am personally not putting increasing test coverage anywhere near high on my priority list ATM. (As stated elsewhere before I would like to put more time into QA, there is just other things more pressing - as always though I suppose.)

**Comment by Kixunil** [comment]
Yeah, I'd be in favor of deactivating the comment at least temporarily while we're doing refactors.

**Comment by apoelstra** [comment]
Same for me. I would also like to increase coverage but I think that'll take an initial concerted effort (and we can do local coverage analysis during that) and the comments don't provide much value until we do that.
